tests/e2e/e2e_utils.py

```diff
@@ -0,0 +1,67 @@
+from datasets import load_dataset
+from transformers import AutoTokenizer
+
+from llmcompressor.modifiers.quantization import GPTQModifier, QuantizationModifier
+from llmcompressor.transformers import SparseAutoModelForCausalLM, oneshot
+from tests.testing_utils import preprocess_tokenize_dataset
+
+
+def run_oneshot_for_e2e_testing(
+    model: str,
+    device: str,
+    oneshot_kwargs: dict,
+    num_calibration_samples: int,
+    max_seq_length: int,
+    dataset_id: str,
+    recipe: str,
+    dataset_split: str,
+    dataset_config: str,
+    scheme: str,
+    quant_type: str,
+):
+    # Load model.
+    loaded_model = SparseAutoModelForCausalLM.from_pretrained(
+        model, device_map=device, torch_dtype="auto"
+    )
+    tokenizer = AutoTokenizer.from_pretrained(model)
+
+    if dataset_id:
+        ds = load_dataset(dataset_id, name=dataset_config, split=dataset_split)
+        ds = ds.shuffle(seed=42).select(range(num_calibration_samples))
+        ds = preprocess_tokenize_dataset(ds, tokenizer, max_seq_length)
+        oneshot_kwargs["dataset"] = ds
+        oneshot_kwargs["max_seq_length"] = max_seq_length
+        oneshot_kwargs["num_calibration_samples"] = num_calibration_samples
+
+    save_dir = model.split("/")[1] + f"-{scheme}"
+
+    oneshot_kwargs["model"] = loaded_model
+    if recipe:
+        oneshot_kwargs["recipe"] = recipe
+    else:
+        # Test assumes that if a recipe was not provided, using
+        # a compatible preset sceme
+        if quant_type == "GPTQ":
+            oneshot_kwargs["recipe"] = GPTQModifier(
+                targets="Linear", scheme=scheme, ignore=["lm_head"]
+            )
+        else:
+            oneshot_kwargs["recipe"] = QuantizationModifier(
+                targets="Linear", scheme=scheme, ignore=["lm_head"]
+            )
+
+    # Apply quantization.
+    print("ONESHOT KWARGS", oneshot_kwargs)
```

Comment:
Oneshot already logs arguments?

Reply:
I find those overwhelming and hard to parse through. This is for so to understand test configuration thats running.

Reply:
Could we use logger.debug instead of print?